test/helpers/messaging_service_helpers.go

```
@@ -180,11 +180,23 @@ func DisconnectMessagingServiceWithFunction(messagingService solace.MessagingSer
 
 // ForceDisconnectViaSEMPv2 function
 func ForceDisconnectViaSEMPv2(messagingService solace.MessagingService) {
+	ForceDisconnectViaSEMPv2WithConfiguration(messagingService, DefaultConfiguration())
+}
+
+// ForceDisconnectViaSEMPv2WithConfiguration function allows different configurations to be provided, such as VPN name
+func ForceDisconnectViaSEMPv2WithConfiguration(messagingService solace.MessagingService, configuration config.ServicePropertyMap) {
+	vpnName, ok := configuration[config.ServicePropertyVPNName].(string)
+	/* NOTE: This interface currently doesn't support returning an error, so if the conversion failed we just set
+	 * the vpn name to the error string and let the test run. This will fail the test and alert the developer that
+	 * something went wrong without needing to change this interface. */
+	if !ok {
+		vpnName = "string conversion for VPN name from service property map failed."
+	}
```

ugly.   Is this the type of thing that leads to strange failures in automation that are hard to debug without calling a developer?

---

@RagnarPaulson , do you still have concerns about this?

---

I don't understand the purpose or the comment about logging in tests, all test frameworks that I know of allow for displaying the reason a test failed,  sometimes through an assert mechanism,  sometimes just by log.  I don't think a test framework would be very useful if it only failed silently.